crates/red_knot_python_semantic/src/types.rs

```diff
@@ -4343,10 +4333,11 @@ impl<'db> FunctionType<'db> {
     ///
     /// Returns `None` if the function is overloaded. This powers the `CallableTypeFromFunction`
     /// special form from the `knot_extensions` module.
-    pub(crate) fn into_callable_type(self, db: &'db dyn Db) -> Option<Type<'db>> {
+    pub(crate) fn into_callable_type(self, db: &'db dyn Db) -> Type<'db> {
         // TODO: Add support for overloaded callables; return `Type`, not `Option<Type>`.
```

Comment:
the TODO seems done?

```suggestion
```

Reply:
I think it's partly done; we still aren't handling `@overload` decorators, which I think are needed to address the first part of the TODO